### DISCUSSION TITLE
Status and future of Stevenarella

### DISCUSSION THREAD

**Discussion by PureTryOut** [iceiix/stevenarella#601] [General]
https://github.com/iceiix/stevenarella/issues/598 spawned a discussion about the future of Stevenarella.

@terrarier2111 seems to have the skills to contribute and is interested, but currently there is just a single developer and it's mostly bumping dependencies rather than getting features working and that is scaring them off. Right now they have forked the project and have made some local, ~~private (:cry:)~~ changes which go in an opposite direction that this project is currently going in. Mainly moving the renderer to Vulkano which gets rid of web support.

**EDIT:** @terrarier2111 has now published his code and it's available on https://github.com/terrarier2111/Leafish

I'm not a fan of splitting efforts, especially when there are only so few people working on it in the first place, and I'd like to prevent forks and have a proper discussion on the future of this project.

@iceiix what are your plans for the future of Stevenarella? Is web support a must have, even if it prevents support for e.g. Vulkan and maybe scares off possible developers? Do you have time to help work on adding features rather than just bumping dependencies all the time?

When anybody responds to discussion, please use the thread function! In other Github discussions in this repo I see people adding new messages separately and context gets lost quickly.

**Comment by PureTryOut** [comment]
Starting a new thread here, one reason people might at the moment not be interested in Stevenarella when they see it is because besides joining servers and walking around, you can't really do much or really "play". I can shout about Stevenarella on some social media platforms (Reddit, Lemmy, Mastodon, etc) which could attract some developers, but longer term it really needs to get to _some_ playable state to attract new users and thus potential developers. That's why I made #598 in the first place; it's in my opinion better to focus on feature compatibility with a specific feature rather than try follow all Mojang updates or add "extras" like web support. Once you have the game playable for at least a single version, you can attract users and developers and start spreading your focus more out on newer versions and awesome features.

**Reply by skillman623** [reply]
I'd definitely try to help out. Currently I'm working on learning how to make GUI's but then I'm planning on starting work on #62.
Is there any group or platform (be it here on github, discord or Mastodon) for communication regarding development of this project?

**Reply by PureTryOut** [reply]
There was/is an IRC channel but it has never really been used much and @iceiix said he doesn't really like instant messaging platforms for such purposes. So I suppose here on Github is the place, both through issues and through these discussion things.

Thanks for wanting to help out! #62 is definitely an important issue and is required for a lot of other stuff so it's good to get that one resolved.

**Reply by terrarier2111** [reply]
> I'd definitely try to help out. Currently I'm working on learning how to make GUI's but then I'm planning on starting work on #62.
> Is there any group or platform (be it here on github, discord or Mastodon) for communication regarding development of this project?

I already did an inventory system with item rendering in my custom version of this project as well as an HUD

**Reply by terrarier2111** [reply]
> There was/is an IRC channel but it has never really been used much and @iceiix said he doesn't really like instant messaging platforms for such purposes. So I suppose here on Github is the place, both through issues and through these discussion things.
> 
> Thanks for wanting to help out! #62 is definitely an important issue and is required for a lot of other stuff so it's good to get that one resolved.

What about a discord server? I think that others might be more responsive on discord than on github and if we want to attract more devs we should some decent way to communicate. I really like the way which the Veloren project is going, they have a discord server and all dev channels in which discussions take place are open to the public (read-only).

**Reply by PureTryOut** [reply]
I uh... Don't want to use Discord. I personally think a FOSS project should not depend on a proprietary communication platform, so my preference would be Matrix or if really needed, IRC. I do agree an instant messaging platform would be good to have but as I said, at least @iceiix doesn't seem to be much interested in it.

**Reply by terrarier2111** [reply]
> I uh... Don't want to use Discord. I personally think a FOSS project should not depend on a proprietary communication platform, so my preference would be Matrix or if really needed, IRC. I do agree an instant messaging platform would be good to have but as I said, at least @iceiix doesn't seem to be much interested in it.

But if you don't want proprietary software for this project, we have to move away from github to gitlab as well.

**Reply by PureTryOut** [reply]
Although that would be great, I find Github less of a problem as at least it's using the FOSS Git tool to do it's things and can be moved away from relatively easily if needed in the future.

But yes alternatives like Gitlab or SourceHut do have my preference :wink:

**Reply by terrarier2111** [reply]
> Although that would be great, I find Github less of a problem as at least it's using the FOSS Git tool to do it's things and can be moved away from relatively easily if needed in the future.
> 
> But yes alternatives like Gitlab or SourceHut do have my preference 😉

But tbh I don't see where Discord is a bigger problem than Github, both are owned by big mega-corps and any data you put on these platforms isn't private anymore

**Reply by terrarier2111** [reply]
Also it would be far easier for the majority of devs to use discord (as most of them already do i suspect) than to move to a new platform alltogether - we could also make some sort of poll and let the majority democratically decide

**Reply by PureTryOut** [reply]
Github is less bad because you can move away from it without losing your development history, and there are 3rd party tools out there to take issues with you as well. But yeah I agree it ain't great either, if the devs are fine with it I would be all for moving to e.g. Gitlab.

> Also it would be far easier for the majority of devs to use discord (as most of them already do i suspect) than to move to a new platform alltogether

You underestimate how many FOSS devs are already using IRC and/or Matrix. If really needed even bridges could be setup to let users choose their own preferred platforms but still have them all talk in the same spot, at least quite a good Matrix<->Discord bridge exists. In the end don't let me block starting up a Discord guild/group/thingy (server is a _really_ shitty name) if that is what everybody wants, but at least I definitely won't join it.

**Reply by terrarier2111** [reply]
> Github is less bad because you can move away from it without losing your development history, and there are 3rd party tools out there to take issues with you as well. But yeah I agree it ain't great either, if the devs are fine with it I would be all for moving to e.g. Gitlab.
> 
> > Also it would be far easier for the majority of devs to use discord (as most of them already do i suspect) than to move to a new platform alltogether
> 
> You underestimate how many FOSS devs are already using IRC and/or Matrix. If really needed even bridges could be setup to let users choose their own preferred platforms but still have them all talk in the same spot, at least quite a good Matrix<->Discord bridge exists. In the end don't let me block starting up a Discord guild/group/thingy (server is a _really_ shitty name) if that is what everybody wants, but at least I definitely won't join it.

If iceix is okay with it, I could manage the dc server and try to bring the teams efforts together, but for the dc server to work it would have to be in the readme of this repo

**Comment by PureTryOut** [comment]
(splitting of to a separate thread to keep context relevant)

@terrarier2111 said:

> I already did an inventory system with item rendering in my custom version of this project as well as an HUD

Any chance you can PR it to here?

**Reply by terrarier2111** [reply]
> (splitting of to a separate thread to keep context relevant)
> 
> @terrarier2111 said:
> 
> > I already did an inventory system with item rendering in my custom version of this project as well as an HUD
> 
> Any chance you can PR it to here?

I could open src my project, but I changed the entire structure of the game (I switched to using Arcs and other concurrency stuff) so it's not really possible to create seperate prs

**Reply by PureTryOut** [reply]
Hmm, ok. At least open-sourcing it would make me happy :wink: 

Maybe @iceiix can have a look at it, maybe they like the direction you're going in and it could replace the current project?

**Reply by terrarier2111** [reply]
> Hmm, ok. At least open-sourcing it would make me happy 😉
> 
> Maybe @iceiix can have a look at it, maybe they like the direction you're going in and it could replace the current project?

I am willing to do that, but I want to fix 2 regressions, which I introduced before replacing the current project.

**Reply by terrarier2111** [reply]
> Hmm, ok. At least open-sourcing it would make me happy 😉
> 
> Maybe @iceiix can have a look at it, maybe they like the direction you're going in and it could replace the current project?

I just published my project, it is called "Leafish", I didn't update it for quite a while, because I am working on a fix for a regression which is a bit complicated

**Reply by terrarier2111** [reply]
As I fixed the regression, I think I am, pretty happy with the current state of Leafish (except for random comments, warnings and todos) but it should but fine for now, how should we proceed @PureTryOut?

**Reply by PureTryOut** [reply]
Wait for @iceiix to have a look at it and in general share their opinion on this discussion and our comments. It's only fair.
